src/addons/synthesis/vivado/generate-macros

```diff
@@ -0,0 +1,27 @@
+#!/bin/bash
+# Generate many Xilinx Verilog blackboxes for SRAM macros.
+# Required parameter: --dcp <folder to dump .dcp files for the macros>
+
+# If you update this file, remember to update src/tools/technology/vivado-srams
+# (TODO: until this is moved into the map phase)
+
+workdir_tag="vivado_macros"
+
+script_dir="$(dirname "$0")"
+source "$script_dir/vivado_common.sh"
+
+if [ -z "$dcp" ]; then
+    >&2 echo "You must set \$dcp to dump .dcp files";
+    exit 1;
+fi
+
+# Clean macro file just in case
+rm -f "$output"
+
+# Gimme my SRAMs, yum yum.
+#$script_dir/generate-macro --sram-module "_plsi_tech_vivado_prebuilt_sram_32b_4096" --width 32 --depth 4096 -o "${workdir}/32b_4096.v" "$vivado" --board_files "$board_files" --dcp "${dcp}/32b_4096.dcp"
+$script_dir/generate-macro --sram-module "_plsi_tech_vivado_prebuilt_sram_32b_2048" --width 32 --depth 2048 -o "${workdir}/32b_2048.v" "$vivado"  --board_files "$board_files" --dcp "${dcp}/32b_2048.dcp"
```

Comment:
For now in order to keep things simple, we simply generate a (small at the moment) list of SRAMs and let the PLSI memory compiler figure it out from there. In the future we could look into custom generating (couldn't figure out the depends script thing).